rmw/include/rmw/ret_types.h

```diff
@@ -26,6 +26,14 @@ typedef int32_t rmw_ret_t;
 #define RMW_RET_OK 0
 #define RMW_RET_ERROR 1
 #define RMW_RET_TIMEOUT 2
+/// The event is not supported by the implementation
+#define RMW_RET_EVENT_UNSUPPORTED 3
+#define RMW_RETCODE_PRECONDITION_NOT_MET 4
+#define RMW_RET_OUT_OF_RESOURCES 5
+#define RMW_RET_NOT_ENABLED 6
```

Comment:
Ok. Check out the usage in in src/ros2/rmw_connext/rmw_connext_shared_cpp/include/rmw_connext_shared_cpp/connext_static_event_info.hpp: should we just return RMW_RET_UNSUPPORTED (will rename as suggested abaove) for these cases?

Reply:
If we're receiving back a NOT_ENABLED I think we just need to convert it to a generic `RMW_RET_ERROR` for now. I don't think it's actually a state we should get into since I think ROS always creates them enabled.

Reply:
It's a bit more involved that just a NOT_ENABLED: just attempting to read the QoS settings through RTI can generate all of the above return types. However, if we want to decouple DDS from RMW then I agree that it makes sense to remove all of the extras (4-9) and simply return RMW_RET_ERROR.